airflow/models/taskmixin.py

```diff
@@ -0,0 +1,72 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from abc import abstractmethod
+
+
+class TaskMixin:
+    """
+    Mixing implementing common chain methods like >> and <<.
+
+    In the following functions we use:
+    Task = Union[BaseOperator, XComArg]
+    No type annotations due to cyclic imports.
+    """
+
+    @abstractmethod
+    def set_upstream(self, other):
```

Comment:
Do we want `set_upstream` to handle things such as `List[XComArg]` or `List[TaskGroup]`? If so, maybe we should add the typing to indicate `other` can be `Union[TaskMixin, List[TaskMixin]]`.

Reply:
I would love to add some type hints, but I'm not sure if using `TaskMixin` is that much informative. But I any other approach will create cyclic imports